internal/pkg/service/buffer/api/dependencies/dependencies.go

```diff
@@ -116,7 +116,7 @@ type forProjectRequest struct {
 
 func NewServerDeps(ctx context.Context, proc *servicectx.Process, cfg config.Config, envs env.Provider, logger log.Logger, tel telemetry.Telemetry) (v ForServer, err error) {
 	ctx, span := tel.Tracer().Start(ctx, "keboola.go.buffer.api.dependencies.NewServerDeps")
-	defer telemetry.EndSpan(span, &err)
+	defer span.End(&err)
```

Comment:
Lepsie to bude vidiet asi na zmenach v kode, napr. tu.

--------------------

Inak, mozeme sa pristavit este pri `defer`:
- Ako vies, registruje to nejaky callback, ktory sa vykona na konci funkcie, LIFO - teda od posledneho definovaného.
- Argumenty funkcie sa ale vyhodnocuju uz **pri definovani** `defer`. Preto je tam `&err` a nie `err`, kedze v tomto momente este nie je ziadna chyba. Preto sa pouzije pointer na premennú, kde na konci bude chyba, definovanu ako return argument `(v ForServer, err error)`.
- Dalo by sa to obist este takto:
```
defer func() {
    span.End(err)
}
```

V tomto pripade  ^^^ nemame ziadny argument a cele sa to vyhodnoti az na konci, ale je to zbytocne dlhy zapis.